File: code_samples/swift/async_promises.swift
import Foundation

/// Async: Promises
/// AI/ML Training Sample

class Promises {
    var data: String
    
    init() {
        self.data = ""
    }
    
    func process(_ input: String) {
        self.data = input
    }
    
    func getData() -> String {
        return self.data
    }
    
    func validate() -> Bool {
        return !self.data.isEmpty
    }
}

// Example usage
let instance = Promises()
instance.process("example")
print("Data: \(instance.getData())")
print("Valid: \(instance.validate())")

File: code_samples/lua/async_promises.lua
-- Async: Promises
-- AI/ML Training Sample

Promises = {}
Promises.__ index = Promises

function Promises:new()
    local obj = {
        data = ""
    }
    setmetatable(obj, self)
    return obj
end

function Promises:process(input)
    self.data = input
end

function Promises:getData()
    return self.data
end

function Promises:validate()
    return #self.data > 0
end

-- Example usage
local instance = Promises:new()
instance:process("example")
print("Data: " .. instance:getData())
print("Valid: " .. tostring(instance:validate()))
